packages/dds/map/src/mapKernel.ts

```diff
@@ -699,6 +699,28 @@ export class MapKernel {
 					// is no optimistically-applied local pending clear that would supersede this remote clear.
 					if (!this.pendingData.some((entry) => entry.type === "clear")) {
 						this.eventEmitter.emit("clear", local, this.eventEmitter);
+
+						// Emit delete-like valueChanged events for keys that did not have pending local operations
+						// and are now removed. This makes the behavior consistent with individual delete operations.
+						// Keys that remain visible due to pending local operations don't get events - consumers
+						// should re-read the map after a clear event to discover retained keys.
+						for (const [key, value] of this.sequencedData) {
```

Comment:
Since we call `this.sequencedData.clear()` above, I think this loop doesn't actually run?  It's correct to emit the events only after clearing the sequenced data, but I think we need to detect which keys need eventing before we `.clear()` it.